charabia/src/segmenter/chinese.rs

```diff
@@ -9,10 +9,57 @@ use crate::segmenter::Segmenter;
 /// without HMM feature.
 pub struct ChineseSegmenter;
 
+fn next_gram<const N: usize>(s: &str) -> Option<&str> {
+    let mut char_count = 0;
+    for (i, _) in s.char_indices() {
+        char_count += 1;
+        if char_count > N {
+            return Some(&s[0..i]);
+        }
+    }
+    if char_count == N {
+        return Some(s);
+    }
+    None
+}
+
+fn cut_for_search(s: &str) -> Vec<&str> {
+    if s.chars().count() <= 2 {
+        return vec![s];
+    }
```

Comment:
This should work and looks better 🤔 
```suggestion
fn cut_for_search<'a>(s: &'a str) -> impl Iterator<Item = &'a str> {
    if s.chars().count() <= 2 {
        return Some(s).into_iter()];
    }
```